Include/cpython/longobject.h

```diff
@@ -4,6 +4,44 @@
 
 PyAPI_FUNC(PyObject*) PyLong_FromUnicodeObject(PyObject *u, int base);
 
+/* PyLong_AsByteArray: Copy the integer value to a native address.
+   n is the number of bytes available in the buffer.
+   Uses the current build's default endianness.
+   PyLong_AsByteArray ensures the MSB matches the int's sign.
+   PyLong_AsUnsignedByteArray allows the MSB to be set for a
+   positive Python value provided no significant bits are lost
+   (that is, all higher bits are a sign extension of the MSB).
+   Negative values still sign extend to fill the buffer and are
+   guaranteed to have MSB set. Use PyLong_AsByteArray to guarantee
+   that the MSB is set iff the int was negative.
+
+   Returns 0 on success or -1 with an exception set, but if the buffer is
+   not big enough, returns the desired buffer size without setting an
+   exception. Note that the desired size may be larger than strictly
+   necessary to avoid precise calculations. */
+PyAPI_FUNC(int) PyLong_AsByteArray(PyObject* v, void* buffer, size_t n);
+PyAPI_FUNC(int) PyLong_AsUnsignedByteArray(PyObject* v, void* buffer, size_t n);
+PyAPI_FUNC(int) PyLong_AsByteArrayWithOptions(PyObject* v, void* buffer,
+                                              size_t n, int options);
+
+#define PYLONG_ASBYTEARRAY_NATIVE_ENDIAN    0x00
+#define PYLONG_ASBYTEARRAY_LITTLE_ENDIAN    0x01
+#define PYLONG_ASBYTEARRAY_BIG_ENDIAN       0x02
```

Comment:
Consider:
- avoiding 0 and 1, so using the similarly named similarly-named `PY_LITTLE_ENDIAN`/`PY_BIG_ENDIAN` will fail
- not using a dedicated value for `PYLONG_ASBYTEARRAY_NATIVE_ENDIAN` (which means flags=0 would fail: there'd be no “default” for `*WithOptions`)

```
#define PYLONG_ASBYTEARRAY_LITTLE_ENDIAN    0x01
#define PYLONG_ASBYTEARRAY_BIG_ENDIAN       0x02
#ifdef Py_LITTLE_ENDIAN
#define PYLONG_ASBYTEARRAY_NATIVE_ENDIAN    PYLONG_ASBYTEARRAY_LITTLE_ENDIAN
#else
#define PYLONG_ASBYTEARRAY_NATIVE_ENDIAN    PYLONG_ASBYTEARRAY_BIG_ENDIAN
#endif
```

Reply:
I want a constant for native endian so that callers who aren't written in C can request it without having to do their own logic, but I'm okay with avoiding 0.

Reply:
(Just fighting with the constants right now, so perfectly timed comment 😄 )